docs/api_structure.rst

```diff
@@ -10,75 +10,83 @@ Fitting 101
-    b_vec = np.random.normal(100, scale=2, size=xdata.shape)
-    ydata = a_vec * xdata + b_vec
+Next level up, if they inherit from ``GradientModel`` then they will have
+``eval_jacobian``, which will numerically evaluate the jacobian of the model.
+Lastly, if they inherit from ``HessianModel``, they will also have
+``eval_hessian`` to evaluate the hessian of the model.
+The standard ``Model`` is all of the above.
 
-And we want to fit it to some model::
+Odd ones out from the current library are ``CallableNumericalModel`` and
+``ODEModel``. They only inherit from ``BaseModel`` and are therefore callable,
+but their other behaviors are custom build.
 
-    a = Parameter('a', value=0, min=0, max=1000)
-    b = Parameter('b', value=0, min=0, max=1000)
-    x = Variable('x')
-    model = a * x + b
-
-If we want to fit this normally (but with a specified minimizer), we'd write
-the following::
-
-    fit = Fit(mode, xdata, ydata, minimizer=BFGS)
-    fit_result = fit.execute()
+Objectives
+----------
 
-Now instead, we want to call the minimizer directly. We first define a custom
-objective function (actually just a chi squared)::
+Objectives wrap both the Model and the data supplied, and expose only the free
+parameters of the model to the outside world.
+When called they must return a scalar. This scalar will be *minimized*, so when
+you need something maximized, be sure to add a negation in the right place(s).
+They can be called by using the parameter names as keyword arguments, or with a
+list of parameter values. The latter is there because this is how ``scipy``
+likes it.
+Be sure to inherit from the abstract base class(es) so you're sure you define
+all the methods that are expected of an objective. Similar to the models, they
+come in three types: ``BaseObjective``, ``GradientObjective`` and
+``HessianObjective``. These must implement ``__call__``, ``eval_jacobian`` and
+``eval_hessian`` respectively.
+
+When defining a new objective, it is best to inherit from ``HessianObjective``
+and to define all three if possible. When feeding a model that is only callable
+to a ``HessianObjective`` no puppies die, ``Fit`` is clever enough to
+prevent this.
 
-    def f(x, a, b):
-        return a * x + b
+Minimizers
+----------
 
-    def chi_squared(a, b):
-        return np.sum((ydata - f(xdata, a, b))**2)
+Last in the chain are the minimizers. They are provided with a function to
+minimize (the objective) and the :class:`~symfit.core.argument.Parameter` s as
+a function of which the objective should be minimized. Note that once again
+there are different base classes for minimizers that take e.g. bounds or
+support gradients. Their :meth:`~symfit.core.minimizers.BaseMinimizer.execute`
+method takes the metaparameters for the minimization.
+Again, be sure to inherit from the appropriate base class(es) if you're
+implementing your own minimizer to make sure all the expected methods are there.
```

Comment:
... to make sure all the expected methods are there, and Fit can figure out what it can do.